website/content/en/docs/advanced-topics/heterogeneous.md

```diff
@@ -0,0 +1,400 @@
+```
+
+If you're planning to distribute the operator via OLM, you can find more information in the [OLM guide for supporting multiarch](https://olm.operatorframework.io/docs/advanced-tasks/ship-operator-supporting-multiarch/).
+
+### Setting the Supported Architectures Operator & Operand Workloads
+[Kubernetes provides a mechanism](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/) called `nodeAffinity` which can be used to limit the possible node targets where a pod can be scheduled. The following example can be used to update a `PodSpec` or `PodTemplateSpec` to prevent the scheduling of pods on nodes of incompatible architectures. Here we compare the `kubernetes.io/arch` key set on the node to ensure that the value matches one of the supported architectures. This assumes that the referenced image points to a manifest list of images for each of the architectures supported. It is important to remember that `nodeAffinity` must be set anywhere a container reference is defined, including `Pod`, `Deployment`, `DaemonSet`, `StatefulSet`, or any other object that defines a `PodSpec` or `PodTemplateSpec`.
+
+```yaml
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/arch
+          operator: In
+          values:
+          - amd64
+          - arm64
+          - ppc64le
+          - s390x
+```
+
+The list of architecture values should only include the architectures the operator supports. The full list of possible `GOARCH` values is [available here](https://github.com/golang/go/blob/master/src/go/build/syslist.go). It should be noted that Kubernetes allows a user to specify both `requiredDuringSchedulingIgnoredDuringExecution` and `preferredDuringSchedulingIgnoredDuringExecution`. The syntax for each block is the same. At a minimum, an operator author should set the required terms since they protect against a pod being scheduled on an architecturally incompatible node. If the operator in question has much better performance on a subset of the supported architectures, it may also be prudent to set `preferredDuringSchedulingIgnoredDuringExecution` so that the optimal arches are selected first if they are available.
+
+#### Integrating with Operator Lifecycle Manager
+
+The Operator Lifecycle Manager (OLM) is often used to distribute operators via operator catalogs. The operator deployment object in an OLM integrated operator is defined in the `ClusterServiceVersion` yaml. It is important to remember to set the node affinity block in each of the `spec.template.spec` under `spec.install.deployments`.
+
+## Examples
+This section attempts to show how each of the sample operators could be updated to declare a set of supported architectures. Most of the time, these changes will be in `yaml` files, but different kinds of operators embed `PodSpec` and `PodTemplateSpec` objects in different places.
+
```

Comment:
In all projects we have the same config by default, see: https://sdk.operatorframework.io/docs/overview/project-layout/.

The only places that we need to configure the node affinity are for the deployment used to run the manager and the sidecar proxy. (no matter the language)

Then, note that for the proxy we can/will set the nodeAffinity by default. See: https://github.com/kubernetes-sigs/kubebuilder/pull/2906/files#diff-cd538f6a8d1b3f24837737051cb800f6ab335924d171ba78db29b8690cd55b46
(so users will not need to update this one)

For the manager, we cannot set it because that depends on how they build the image. But we will add the code commented to try encourage and help them out see: https://github.com/kubernetes-sigs/kubebuilder/pull/2906/files#diff-72ce7e8fc292f0c680218ed1cfce019d7baa9c0e41218b8526eecfd57d0bf344

Also, in the same PR, we are adding a new target to help them out to build the images using buildx.